src/redux/asyncThunks/postThunk.js

```diff
@@ -47,4 +47,64 @@ const dislikePost = createAsyncThunk(
   },
 )
 
-export { getPost, likePost, dislikePost }
+const createPost = createAsyncThunk(
+  'posts/createPost',
+  async ({ postData, token }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        '/api/posts',
+        { postData },
+        {
+          headers: { authorization: token },
+        },
+      )
+      const data = { data: response.data }
+      // console.log(response)
+      return data
+    } catch (error) {
+      return rejectWithValue({
+        data: error.response.data,
+      })
+    }
+  },
+)
+
+const deletePost = createAsyncThunk(
+  'posts/deletePost',
+  async ({ _id, token }, { rejectWithValue }) => {
+    try {
+      const response = await axios.delete(`/api/posts/${_id}`, {
+        headers: { authorization: token },
+      })
+      const data = { data: response.data, status: response.status }
+      return data
+    } catch (error) {
+      return rejectWithValue({
+        data: error.response.data,
+        status: error.response.status,
+      })
+    }
+  },
+)
+
+const editPost = createAsyncThunk(
+  'posts/editPost',
+  async ({ postDetail, postData, token }, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(
+        `/api/posts/edit/${postDetail._id}`,
+        { postData },
+        { headers: { authorization: token } },
+      )
+      console.log(response,"edit")
+      const data = { data: response.data, status: response.status }
+      return data
+    } catch (error) {
+      console.error(error)
```

Comment:
this too

Reply:
here I need console to check the error message that's why I used